plugins/restapi/v1/messages.go

```diff
@@ -168,8 +169,13 @@ func messageIDsByIndex(c echo.Context) (*messageIDsByIndexResponse, error) {
 		return nil, errors.WithMessage(restapi.ErrInvalidParameter, "query parameter index empty")
 	}
 
+	indexBytes, err := hex.DecodeString(index)
```

Comment:
If some pleb passes `index` longer than 64 bytes, then the `GetIndexMessageIDs` will call `PadIndexationIndex`, which will call a `make` with a negative count...
```go
// PadIndexationIndex returns a padded indexation index.
func (index []byte) []byte {
	return append(index, make([]byte, IndexationIndexLength-len(index))...)
}``